Sources/firebase/include/FirebaseFirestore.hh

```diff
@@ -48,6 +48,32 @@ inline ::firebase::firestore::DocumentReference
 collection_document(::firebase::firestore::CollectionReference collection, std::string document_path) {
   return collection.Document(document_path);
 }
+
+typedef void (*SnapshotListenerTypedCallback_SwiftWorkaround) (const ::firebase::firestore::DocumentSnapshot *snapshot,
+                                                               ::firebase::firestore::Error *error_code,
+                                                               const char *error_message,
+                                                               void *user_data
+                                                               );
```

Comment:
I think that we should start working to extract the changes that we need in the firebase-cpp-sdk to support Swift (e.g. the future workaround).  Those we can then start to float upstream.